libsignal-service/src/push_service.rs

```diff
@@ -112,25 +112,38 @@ pub struct DeviceCapabilities {
 
 pub struct ProfileKey(pub [u8; 32]);
 
+impl ProfileKey {
+    pub fn derive_access_key(&self) -> Vec<u8> {
+        let key = GenericArray::from_slice(&self.0);
+        let cipher = Aes256Gcm::new(key);
+        let nonce = GenericArray::from_slice(&[0u8; 12]);
+        let buf = [0u8; 16];
+        cipher.encrypt(nonce, &buf[..]).unwrap()
+    }
+}
+
 #[derive(Debug, Deserialize, Default)]
 #[serde(rename_all = "camelCase")]
 pub struct PreKeyStatus {
     pub count: u32,
 }
 
 #[derive(Clone)]
-pub struct HttpCredentials {
+pub struct HttpAuth {
     pub username: String,
     pub password: String,
 }
 
-impl ProfileKey {
-    pub fn derive_access_key(&self) -> Vec<u8> {
-        let key = GenericArray::from_slice(&self.0);
-        let cipher = Aes256Gcm::new(key);
-        let nonce = GenericArray::from_slice(&[0u8; 12]);
-        let buf = [0u8; 16];
-        cipher.encrypt(nonce, &buf[..]).unwrap()
```

Comment:
Something weird happened to the diff here, mind fixing that up?

Reply:
Done!